vivisect/__init__.py

```diff
@@ -958,11 +958,16 @@ def parseOpcode(self, va, arch=envi.ARCH_DEFAULT):
     def iterJumpTable(self, startva):
         ptrbase = startva
         rdest = self.castPointer(ptrbase)
-        while self.isValidPointer(rdest):
+        while self.isValidPointer(rdest) and self.analyzePointer(rdest) not in (LOC_STRING, LOC_UNI):
+            if self.analyzePointer(ptrbase) in (LOC_STRING, LOC_UNI):
```

Comment:
I played around with just making it just `self.analyzePointer(rdest) == LOC_OP`. I'm not sure which is better automated analysis-wise. My gut says to change it to LOC_OP since that would be the most technically accurate. But in terms of reality, I'm not so sure.

Thoughts?

Reply:
Scratch that, unittests fail if I do just ` == LOC_OP`, so something else is needed.

Reply:
i'm still concerned about LOC_POINTER, especially for user-initiated makeFunction() after initial analysis (meaning pointer-analysis has completed through autoanalysis.  if we run into code that has accidentally been made into a pointer, i don't want to fail.  yes, i know we go for "big failures".  unfortunately that doesn't apply well here, since it seriously hinders human analysis, and you never know what opcodes are going to look like pointers.

Reply:
how about making a tuple of "stopper locs" and just reference it here in both of these (the while and the subsequent if)?

Reply:
(Side note: I think I also missed a change on line 965). But good points.
So something like this?
```
STOP_LOCS = (LOC_STRING, LOC_UNI, LOC_STRUCT, LOC_CLSID, LOC_VFTABLE, LOC_IMPORT, LOC_PAD, LOC_NUMBER)
        while self.isValidPointer(rdest) and self.analyzePointer(rdest) in (None, LOC_OP):
            if self.analyzePointer(ptrbase) in STOP_LOCS:
```